db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/StaticMethodCallNode.java

```diff
@@ -222,54 +227,36 @@ public JavaValueNode bindExpression(FromList fromList,
             // look for a routine
 
             String schemaName = procedureName.getSchemaName();
-
-            boolean noSchema = schemaName == null;
-
-            SchemaDescriptor sd = getSchemaDescriptor(schemaName, schemaName != null);
-
-            // The field methodName is used by resolveRoutine and
-            // is set to the name of the routine (procedureName.getTableName()).
-            resolveRoutine(fromList, subqueryList, aggregateVector, sd);
-
-            // (Splice)    This logic, to implicitly check for routine using SYSFUN schema,
-            // was moved here from a few lines down, so that it has a chance to find
-            // aggregate functions in SYSFUN catalog too, such that it will get
-            // a resolvedAggregate. There was no clean way to avoid this Derby fork,
-            // because there's no clean way to tell the parser to use a Splice override
-            // of StaticMethodCallNode.
-
-            if (ad == null && noSchema && !forCallStatement)
-            {
-                // Resolve to a built-in SYSFUN function but only
-                // if this is a function call and the call
-                // was not qualified. E.g. COS(angle). The
-                // SYSFUN functions are not in SYSALIASES but
-                // an in-memory table, set up in DataDictionaryImpl.
-                sd = getSchemaDescriptor("SYSFUN", true);
-
-                resolveRoutine(fromList, subqueryList, aggregateVector, sd);
-                setIsSystemFunction(true);
+            Object candidateSchemasProperty = getLanguageConnectionContext()
+                    .getSessionProperties()
+                    .getProperty(SessionProperties.PROPERTYNAME.CURRENTFUNCTIONPATH);
+            SchemaDescriptor sd = null;
+            if(candidateSchemasProperty != null) {
+                String[] candidateSchemas = (String[])(candidateSchemasProperty);
+                for(String candidateSchema : candidateSchemas) {
+                    try {
+                        sd = getSchemaDescriptor(candidateSchema, candidateSchema != null);
```

Comment:
Is candidateSchema != null always true? A cleaner approach is not to raise exception and then check whether the schema exists
sd = getSchemaDescriptor(candidateSchema, false);
if (sd == null)
  throw StandardException.newException(SQLState.LANG_CURRENT_FUNCTION_PATH_SCHEMA_DOES_NOT_EXIST,Arrays.toString(candidateSchemas),candidateSchema);

Reply:
Good point, thanks I fixed it.